projects/plugins/boost/app/modules/cache/Boost_File_Cache.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Page_Cache;
+
+require_once __DIR__ . '/Boost_Cache.php';
+
+class Boost_File_Cache extends Boost_Cache {
+	/*
+	 * Serve the cached page if it exists, otherwise start output buffering.
+	 */
+	public function serve() {
+		if ( ! $this->get() ) {
+			$this->ob_start();
+		}
+	}
+
+	/*
+	 * Returns the path to the cache directory for the given request, or current request
+	 *
+	 * @param string $request_uri - The request uri to get the path for. Defaults to current request.
+	 * @return string - The path to the cache directory for the given request.
+	 */
+	private function path( $request_uri = false ) {
+		if ( $request_uri !== false ) {
+			$request_uri = $this->sanitize_request_uri( $request_uri );
+		} else {
+			$request_uri = $this->request_uri;
+		}
+
+		$key  = $this->path_key( $request_uri );
+		$path = WP_CONTENT_DIR . DIRECTORY_SEPARATOR . 'boost-cache' . DIRECTORY_SEPARATOR . 'cache' . DIRECTORY_SEPARATOR;
+		for ( $i = 0; $i <= 5; $i++ ) {
+			$path .= substr( $key, $i, 1 ) . DIRECTORY_SEPARATOR;
+		}
+
+		return $path;
+	}
+
+	/*
+	 * Returns the cache filename for the given request, or current request.
+	 *
+	 * @param array $args - an array containing the request_uri, cookies array, and get array representing the request.
+	 * @return string - The cache path + filename for the given request.
+	 */
+	private function cache_filename( $args = array() ) {
+		$defaults = array(
+			'request_uri' => $this->request_uri,
+			'cookies'     => $_COOKIE, // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			'get'         => $_GET, // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.NonceVerification.Recommended
+		);
+		$args     = array_merge( $defaults, $args );
+
+		return $this->path( $args['request_uri'] ) . $this->cache_key( $args ) . '.html';
+	}
```

Comment:
This passes the $args object down to `$this->cache_key`, which will receive the arguments and then re-sanitize them. `$this->request_uri` will already be sanitized on class construction. Is there any danger to sanitizing the request_uri twice, or will it always result in the same value?

Reply:
It's safe to normalize the value twice, definitely since I removed the deep_replace. This should be ok, but I'll add a check here, if it's a different request_uri, then normalize it.